src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,163 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node;
+        if (size == 0) {
+            node = new Node<T>(null, value, null);
+            tail = node;
+            head = node;
+        } else {
+            node = new Node<T>(tail, value, null);
+            tail.next = node;
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexAddByIndex(index);
+        if (index == size) {
+            add(value);
+            return;
+        }
```

Comment:
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        checkIndex(index);
```
If we call the checkIndex(index); method after the first check, we don't need the checkIndexAddByIndex method